torch/csrc/jit/interpreter.cpp

```diff
@@ -412,6 +412,21 @@ struct CodeImpl {
     for (auto node : block->nodes()) {
       const auto& source_location = node->getSourceLocation();
       switch (node->kind()) {
+        case prim::profile:
```

Comment:
do we need profiling options to be in the IR itself? can we get away with a simple map/set of Values with profiling options?

Reply:
We can, but it ends up being a way more complicated and way less flexible. I have another branch where I explored that option. Interpreter runs a *different* graph from the one we want to collect profiling info (stored in an `ExecutionPlan`), so we will need to keep a map from the values in a new copied graph back to the values in the original graph whereas with profile instructions we have a flexibility to use the old value ids or (any kind of integral identifier for that matter) directly. They also have an added benefit of interfering with optimizations, so we know that values won't disappear on us, although this is less of an issue as of now since we are profiling the optimized version of a graph.